### PR TITLE
Fix shared service object created multiple times in cosmos

### PR DESCRIPTION
**Pull request by tanya-borisova**
# PR for issue https://github.com/microsoft/AzureTRE/issues/1703

## What is being addressed

Currently, there is a bug when a shared service object is created multiple times in Cosmos (all these objects are "active", i.e. isActive is set to true).
Only one shared service of a given template type can exist at the time. This PR implements this constraint.

## How is this addressed

- Fixed API and database repository for shared services to return 409 Conflict when there already is an active shared service of a given name.
- DB migration script that deletes all duplicated items

## NOTE

After this PR goes in, everybody who has TRE deployed *needs* to run the DB migration script, otherwise consecutive operations to shared services will fail.

The script can be run as follows:

```
STATE_STORE_ENDPOINT=https://cosmos-${TRE_ID}.documents.azure.com:443/ STATE_STORE_KEY=your_secret_here COSMOSDB_DATABASE_NAME=AzureTRE python3 scripts/db_migrations.py
```

### CODE REVIEW

File: api_app/db/repositories/shared_services.py [hunk 1]
```diff
@@ -26,6 +27,10 @@ def shared_service_query(shared_service_id: str):
     def active_shared_services_query():
         return f'SELECT * FROM c WHERE {IS_ACTIVE_CLAUSE} AND c.resourceType = "{ResourceType.SharedService}"'
 
+    @staticmethod
+    def active_shared_service_with_template_name_query(template_name: str):
+        return f'SELECT * FROM c WHERE {IS_ACTIVE_CLAUSE} AND c.resourceType = "{ResourceType.SharedService}" AND c.templateName = "{template_name}"'
```
Comment (ross-p-smith) [inline api_app/db/repositories/shared_services.py:32]: Do we need the template version in here too?

**Review by ross-p-smith — APPROVED** [review]
Looks great, just one question about whether we need to check the version too?

File: api_app/db/repositories/shared_services.py (hunk 1 above)
Comment (tanya-borisova) [reply api_app/db/repositories/shared_services.py:32]: This is a great question that I should've clarified explicitly :) 

If we had a template version here, that would mean that we allow several active shared services of the same resource type, but of different version.

Currently, I think we don't want to allow that (cc @damoodamoo  @marrobi), and instead we want to implement an upgrade action, which would upgrade an already existing shared service to a new version (see https://github.com/microsoft/AzureTRE/issues/141). 

So, currently, once a shared service is deployed, it cannot then be upgraded to a new version. This limitation will be solved as part of that issue I linked above

**Comment by tanya-borisova** [comment]
I think https://github.com/microsoft/AzureTRE/pull/1726 should go in first, and I would propose to add running db-migrations as part of `make all`. Otherwise, as it currently stands, if people don't run migrations first, then consecutive deployments of shared services including firewall will fail with 500 Internal server error (as the code checks for invariant that only one shared service is active at a time).

**Pull request merged**